File: src/main/java/io/poshidaev/toda/entity/Topic.java
package io.poshidaev.toda.entity;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import java.util.List;

@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Topic extends BaseEntity {

    public Topic(String text) {

        this.text = text;
    }

    @Id
    private String text;

    @ManyToMany(mappedBy = "topics", fetch = FetchType.LAZY)
    private List<Goal> goals;
}

File: src/main/kotlin/io/poshidaev/toda/entity/Topic.kt
package io.poshidaev.toda.entity

import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.Id
import javax.persistence.ManyToMany

@Entity
class Topic : BaseEntity {

    constructor(text: String?) : super() {
        this.text = text
    }

    constructor() : super()


    @Id
    var text: String? = null

    @ManyToMany(mappedBy = "topics", fetch = FetchType.LAZY)
    var goals: List<Goal>? = mutableListOf()
}